packages/nocodb-sdk/src/lib/filterHelpers.ts

```
@@ -0,0 +1,320 @@
+    return extractCondition(
+      nestedArrayConditions || [],
+      aliasColObjMap,
+      throwErrorIfInvalid
+    );
+  }
+
+  // iterate until finding right closing
+  while (
+    (nextOpenIndex = str
+      .substring(0, closingIndex)
+      .indexOf('((', nextOpenIndex + 1)) != -1
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)